src/scripts/checkheader.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+# *
+# * This Source Code Form is subject to the terms of the Mozilla Public
+# * License, v. 2.0. If a copy of the MPL was not distributed with this
+# * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+# *
+
+echo "[MPL] Checking source files for MPL headers..."
+
+# Checks for the existance of the MPL header
+MPL_REGEX=".*?This\W+Source\W+Code\W+Form\W+is\W+subject\W+to\W+the\W+terms\W+of\W+the\W+Mozilla\W+Public\W+License,\W+v\.\W+2\.0\.\W+If\W+a\W+copy\W+of\W+the\W+MPL\W+was\W+not\W+distributed\W+with\W+this\W+file,\W+You\W+can\W+obtain\W+one\W+at\W+http:\/\/mozilla\.org\/MPL\/2\.0\/\."
+
+# This find commands lists all our source code files, excluding bad directories
+# It prunes gitignore and gitattributes, and everything in the public dir, .git, node_modules, and postgres-data
+FILES=$(find . -type f \( -iname '*.jsx' -o -iname '*.js' -o -iname '*.sh' -o -iname '*.sql' -o -iname "*.y*ml" \
```

Comment:
Could this be slightly generalized to a list of files/paths we want to ignore (like a .gitignore)?
It seems tricky to add additional parameters to the regex.

Reply:
I would also prefer this solution. I think it would be nice to have piecemeal control over exactly which files we want excluded from the header check.

Reply:
It's not a regex, it's just a find command, which basically says: include everything which matches the first line, which is a file, and which doesn't match anything on the second two lines. 

There's not really a better way to do this without introducing additional dependencies; we could theoretically write the script in JS, I suppose, but I really don't think this is that complex.